tidymodels-intro-part2.qmd

```diff
@@ -0,0 +1,232 @@
+
+```{r}
+true_death_rates <- case_death_rate_subset %>% 
+  filter(geo_value %in% c("ca", "ny", "ar", "tx", "hi") &
+      time_value == "2021-03-15") %>% 
+  select(geo_value, death_rate) 
+```
+
+To adjoin the two dataframes containing the truth and the predictions, we could force `bind_cols()` to work, but one important thing is that the order of the rows (states) should match up in both otherwise we could mismatch data (meant for one state to another). For instance, in our example notice that the first `geo_value` in `true_death_rates` is for `ar`, while it is for `ak` in `p_round2`. And if we had more states to deal with, it could be especially troublesome to have to go in and re-organize the rows. In such situations, we can turn to `dplyr`'s `left_join()` function to add `true_death_rates` values as a column to the `p_round2` dataframe by `geo_value`. The order of the rows doesn't matter for this function because it matches based on the linking variable `geo_value`.
+
+```{r}
+p_w_truth <- left_join(p_round2, true_death_rates, by = "geo_value")
+p_w_truth
+```
+
+Now let's use `metrics()` function to briefly assess our model performance. 
+
+```{r}
+p_w_truth %>%
+  metrics(truth = death_rate, estimate = .pred)
+```
+
+As before, let's briefly go through the metrics that were produced. Root mean square error (RMSE) is the square root of the average squared difference between the predicted and the actual values. So it is a measure of accuracy because it informs us how well the model hit the mark (the truth). Since this is an error, lower values are better as you might expect, and the absolute minimum value that could be achieved is 0. 
+
+Mean absolute error (MAE) is the average absolute difference between the predicted and the actual values. It's interpretation is similar to RMSE in that lower values are better and the minimum that can be achieved is also 0. A major difference between them is how much each error contributes. RMSE is more sensitive to outliers (just think of what happens when you square a large value - ie. a large difference between the predicted and the truth), so the impact of larger errors is magnified, while smaller errors don't contribute as much. In contrast, the error contribution in MAE increases linearly (ex. an error of 4 would contribute twice as much as an error of 2). 
+
+R-squared (Rsq) or the coefficient of determination tells us what proportion variation of the response is explained or captured by the independent variables. So if R-squared is 0.7, this means that the predictors explain about 70% of the variability in the response. It's calculated as the squared correlation coefficient, so it also ranges between 0 and 1, and being closer to 1 (ie. explaining more of the variance in the response) is better.
```

Comment:
I hate R2. Let's remove it completely.

https://ubc-stat.github.io/stat-406/lectures/slides/05-estimating-test-mse.html#4